contracts/Controller.sol

```diff
@@ -47,25 +51,27 @@ contract Controller is Ownable {
     /**
      * @notice modifier check if protocol is not paused
      */
-    // modifier isNotPaused {
-    //     _;
-    // }
+    modifier isNotPaused {
+        require(!systemPaused, "Controller: system is paused");
 
-    /**
-     * @notice modifier to check if otoken is expired
-     * @param _otoken otoken address
-     */
-    // modifier isExpired(address _otoken) {
-    //     _;
-    // }
+        _;
+    }
 
     /**
-     * @notice modifier to check if sender is an authorized vault operator
+     * @notice modifier to check if sender is an account owner or an authorized account operator
      * @param _sender sender address
+     * @param _accountOwner account owner address
      */
-    // modifier isAuthorized(address _sender) {
-    //     _;
-    // }
+    modifier isAuthorized(address _sender, address _accountOwner) {
+        MarginAccount.Account memory senderAccount = accounts[_sender];
+
+        require(
+            (senderAccount.vaultCounter != 0) || (operators[_accountOwner][_sender]),
```

Comment:
why do we need to check senderAccount.vaultCounter? What would happen if this is the first operation?

Reply:
Updated this.